src/Microsoft.AspNet.JsonPatch/Adapters/ObjectAdapter.cs

```diff
@@ -636,11 +637,14 @@ public void Copy(Operation<T> operation, T objectToApplyTo)
             }
         }
 
-        private bool IsNonStringArray(JsonPatchProperty patchProperty)
+        private bool IsNonStringArray(Type type)
         {
-            return !(patchProperty.Property.PropertyType == typeof(string))
-                    && typeof(IList).GetTypeInfo().IsAssignableFrom(
-                        patchProperty.Property.PropertyType.GetTypeInfo());
+            if (PropertyHelpers.GetIListType(type) != null)
```

Comment:
`GetIListType` is used in multiple places in this class. I just updated this code to re-use existing method.